app/models/git_repository.rb

```diff
@@ -155,6 +183,36 @@ def handling_worktree_errors
     raise MiqException::Error, err.message
   end
 
+  def git_lock_filename
+    @git_lock_filename ||= LOCKFILE_DIR.join(id.to_s)
+  end
+
+  def acquire_git_lock
+    return false if defined?(@git_lock) && !@git_lock.nil?
+
+    FileUtils.mkdir_p(LOCKFILE_DIR)
+
+    @git_lock = File.open(git_lock_filename, File::RDWR | File::CREAT, 0o644)
+    @git_lock.flock(File::LOCK_EX)                         # block waiting for lock
+    @git_lock.write("#{Process.pid} - #{Time.zone.now}\n") # for debugging
```

Comment:
If we don't unlink the file later, will this just keep appending since you've opened RDWR?

Reply:
All usages of the this have their cursors at 0, so when they are writing, they are each starting at 0, and not appending on top of each other.

See https://github.com/ManageIQ/manageiq/pull/19074/files#r309950154